### PR TITLE
Run some pfstats task only on the Master of the cluster or on the standalone

### LINKED ISSUE #3690: Some task from pfstat should only run on the server who have the vip
We have duplicate sql queries on te database coming from pfstats.
We will need to disable these queries from the servers who don't own the vip.

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Get rid of sql queries made by pfstats on a cluster

# Impacts
Stats

# Issue
fixes #3690

# Delete branch after merge
YES

# NEWS file entries

## Enhancements
* pfstats optimization

## Bug Fixes
* Run some sql queries from pfstats from only one server (#3690)

### CODE REVIEW

File: go/stats/main.go [hunk 1]
```diff
@@ -1764,3 +1781,38 @@ var Verb = TypeName{
 		},
 	},
 }
+
+// Detect the vip on management
+func detectVIP(management pfconfigdriver.ManagementNetwork) {
```
Comment (julsemaan) [inline go/stats/main.go:1786]: This seems duplicated with the one in pfdhcp, could we centralize this in a package so we don't duplicate code

Comment (fdurand) [reply go/stats/main.go:1786]: It's not exactly the same that we have in pfdhcp.
Btw i can move this function in sharedutils.

File: conf/stats.conf.example [hunk 2]
```diff
@@ -7,6 +7,7 @@ api_path=/api/v1/queues/stats
 api_compile=$.items[*].queue, $.items[*].stats.count
 interval=60s
 randomize=true
+management = false
```
Comment (satkunas) [inline conf/stats.conf.example]: Use the same `key=value` spacing used elsewhere.

File: go/stats/main.go [hunk 3]
```diff
@@ -1764,3 +1781,38 @@ var Verb = TypeName{
...
+	if keyConfCluster.Ip == "" {
+		VIP[management.Int] = true
+		return
+	}
+
+	eth, _ := net.InterfaceByName(management.Int)
+	adresses, _ := eth.Addrs()
+	var found bool
+	found = false
+	for _, adresse := range adresses {
```
Comment (satkunas) [inline go/stats/main.go:1801]: This `for` loop appears to iterate over all `adresses` even after `found=true`. Is a `return` statement needed right after `found=true`?

Would it be faster to define `VIP[management.Int] = false` at the beginning of this function, and only set `VIP[management.Int] = true` and `return` immediately if `IP.Equal(VIPIp[management.Int])` ?

**Review by satkunas — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
BUMP

**Comment by julsemaan** [comment]
@satkunas 

Can you check @fdurand's changes and remove your block if its all fine.

To me, we're all good to merge

**Comment by satkunas** [comment]
@fdurand @julsemaan all good. Merge when ready.

**Pull request merged**